### ISSUE TITLE
webpack generates sourcemaps with invalid sources

### ISSUE THREAD

**Issue by dgoldstein0** [webpack/webpack#4889]
**Do you want to request a *feature* or report a *bug*?**

Bug

**What is the current behavior?**

I'm creating an AMD bundle of react-test-renderer so we can load react-test-render with requirejs.  During this, I discovered that webpack is generating some things in the `sources` field of the sourcemap which don't correspond to files on disk - namely
a) my externals show up as sources
b) there's a mysterious `"webpack/bootstrap 50756633eb024ca71d38"` which I don't know where it comes from but is definitely not a file anywhere

**If the current behavior is a bug, please provide the steps to reproduce.**

my webpack config:
```
var path = require('path');
var outputDir = '../../../compiled/js/external';

module.exports = {
  entry: require.resolve('react-test-renderer'),
  output: {
    filename: 'react-test-renderer.js',
    path: outputDir,
    libraryTarget: 'amd',
    devtoolModuleFilenameTemplate: function (info) {
      // replace webpack:// urls with relative paths
      return path.relative(outputDir, info.absoluteResourcePath);
    },
  },
  devtool: 'source-map',

  externals: [
    {
      'react/lib/React': 'react',
    },
  ],
};
```

package.json:

```
{
  "scripts": {
    "build": "webpack --optimize-minimize",
    "build-dev": "webpack"
  },
  "dependencies": {
    "react": "^15.4.2",
    "react-test-renderer": "^15.4.2"
  },
  "devDependencies": {
    "webpack": "^2.5.1"
  }
}
```

This ends up generating a mostly valid sourcemap, but with two `sources` that aren't actual files:
"../../../src/external/react-test-renderer/webpack/bootstrap 50756633eb024ca71d38"
"../../../src/external/react-test-renderer/external \"react\""

**What is the expected behavior?**

everything in the sources field should be a file on disk, that if resolved correctly relative to sourceRoot of the sourcemap, can be read from disk.

I've tried using SourceMapDevToolPlugin instead, to get finer grained control of sourcemap generation, but I can't seem to get it to do anything different to either `external "react"` or `webpack/bootstrap 50756633eb024ca71d38`.

**Please mention other relevant information such as the browser version, Node.js version, webpack version and Operating System.**

fwiw I'm using some ancient things (node 0.12.7, ubuntu 12.04) but I don't think they are the source of this bug.  I upgraded from webpack 1.13 -> 2.5.1 while trying to work through this and it's the same in both.

**Comment by sokra** [comment]
cc @timse

**Comment by dgoldstein0** [comment]
correction - I actually encountered this with webpack 1.13 - just realized that since I had a shrinkwrap file sitting around that I never actually installed 2.5.1, and my ancient version of nodejs doesn't work with the new webpack.  I tried to test with a newer node so I could use the latest webpack, but couldn't get that working easily on my machine.  So if someone could repro with the latest node, that would be useful data.
